### PR TITLE
Adding Vivillon Patterns

### PR DESCRIPTION
**Pull request by NiCeDiCe90**
Adds all Vivillon Pattern sprites into the game
new Vivillon Questline
new Yearly Event

<3

Big THANKS to Jessica for the help.

### CODE REVIEW

**Comment by NiCeDiCe90** [comment]
Big Thanks to PixL!
Quest works fine now. 👍

File: src/scripts/gym/Gym.ts [hunk 1]
```diff
@@ -1104,9 +1105,11 @@ gymList['Snowbelle City'] = new Gym(
     BadgeEnums.Iceberg,
     52000,
     'Impressive! Your Pokémon fought with great courage. I can tell that you\'ve trained your Pokémon well.',
-    [new ClearDungeonRequirement(1, GameConstants.getDungeonIndex('Pokémon Village'))]
+    [new ClearDungeonRequirement(1, GameConstants.getDungeonIndex('Pokémon Village'))],
+    () => {
+        App.game.quests.getQuestLine('The great hunt').beginQuest();
```
Comment (RedSparr0w) [inline src/scripts/gym/Gym.ts:1110]: Shouldn't this be `The Great Vivillon Hunt!`?

Comment (NiCeDiCe90) [reply src/scripts/gym/Gym.ts]: Yes. I'm very sorry. I overlooked it and forgot to change the Name there.
I did it now. 😭

**Comment by RedSparr0w** [comment]
Are the HP amounts supposed to be `93,659,450` for all of them, or was that just a placeholder?

**Comment by NiCeDiCe90** [comment]
> Are the HP amounts supposed to be `93,659,450` for all of them, or was that just a placeholder?

It's supposed to be this high.
The bosses of the most difficult dungeon in the quest have this amount of HP. 
Vivillon is a gen 6 Pokémon and I don't want it to be too easy to catch in early game dungeons so it will be very bulky if you use chests in easy dungeons.

**Review by RedSparr0w — APPROVED** [review]
Everything looks good to me.

I've just reduced the sprite file sizes and reduced some duplication in the code.

Can you just double check I've re-added all the Vivillon correctly and that everything is still working for you?

I gave it a quick run through of the first couple and all seems to be working the same as before.

**Comment by NiCeDiCe90** [comment]
Wow that's insane effort you've put into the quest. Thank you very much!
It looks all right to me. The Vivillon and dungeons seem to be all right. 
I can test it later today when I'm back home (in about 9-10h from now). But if you say the Questline works I don't think I will find any new issues. 
I hope the hints are ok. English is not my native language so I wasn't sure. Feel free to change them if they are wrong or sound weird.

<3

**Comment by NiCeDiCe90** [comment]
So I tested the build and I can't find anything that doesn't go as expected.
Every cure litte Butterfly sits at his place.

Thanks again. Can't wait to see them in the live build :3

**Pull request merged**